packages/core/admin/admin/src/components/AuthenticatedApp.tsx

```diff
@@ -1,58 +1,94 @@
-import React, { useEffect, useState } from 'react';
+import * as React from 'react';
 
 import {
   AppInfoProvider,
   auth,
   LoadingIndicatorPage,
+  useFetchClient,
   useGuidedTour,
-  useNotification,
 } from '@strapi/helper-plugin';
-import get from 'lodash/get';
+import lodashGet from 'lodash/get';
 import { useQueries } from 'react-query';
+import lt from 'semver/functions/lt';
+import valid from 'semver/functions/valid';
 //  TODO: DS add loader
 
-import packageJSON from '../../../../package.json';
-import { useConfiguration } from '../../hooks/useConfiguration';
-import { getFullName, hashAdminUserEmail } from '../../utils';
-import { NpsSurvey } from '../NpsSurvey';
-import { PluginsInitializer } from '../PluginsInitializer';
-import RBACProvider from '../RBACProvider';
+import packageJSON from '../../../package.json';
+import { useConfiguration } from '../hooks/useConfiguration';
+// @ts-expect-error - no types yet.
+import { getFullName, hashAdminUserEmail } from '../utils';
 
-import { fetchAppInfo, fetchCurrentUserPermissions, fetchUserRoles } from './utils/api';
-import { checkLatestStrapiVersion } from './utils/checkLatestStrapiVersion';
-import { fetchStrapiLatestRelease } from './utils/fetchStrapiLatestRelease';
+import { NpsSurvey } from './NpsSurvey';
+import { PluginsInitializer } from './PluginsInitializer';
+import { RBACProvider, Permission } from './RBACProvider';
 
 const strapiVersion = packageJSON.version;
 
 const AuthenticatedApp = () => {
   const { setGuidedTourVisibility } = useGuidedTour();
-  const toggleNotification = useNotification();
-  const userInfo = auth.getUserInfo();
-  const userName = get(userInfo, 'username') || getFullName(userInfo.firstname, userInfo.lastname);
-  const [userDisplayName, setUserDisplayName] = useState(userName);
-  const [userId, setUserId] = useState(null);
+  const userInfo = auth.get('userInfo');
+  const userName = userInfo
+    ? lodashGet(userInfo, 'username') || getFullName(userInfo.firstname, userInfo.lastname)
+    : null;
+  const [userDisplayName, setUserDisplayName] = React.useState(userName);
+  const [userId, setUserId] = React.useState(null);
   const { showReleaseNotification } = useConfiguration();
+  const { get } = useFetchClient();
   const [
     { data: appInfos, status },
     { data: tagName, isLoading },
     { data: permissions, status: fetchPermissionsStatus, refetch, isFetching },
     { data: userRoles },
   ] = useQueries([
-    { queryKey: 'app-infos', queryFn: fetchAppInfo },
+    {
+      queryKey: 'app-infos',
+      async queryFn() {
+        const { data } = await get('/admin/information');
```

Comment:
Do we not want to infer the response shape here?

Reply:
yes! thanks for the reminder, I thought we had `noImplicitAny` on to stop this 🤔